lib/segment/src/index/hnsw_index/gpu/gpu_insert_context.rs

```diff
@@ -182,25 +182,29 @@ impl<'a> GpuInsertContext<'a> {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
         gpu_vector_storage: &'a GpuVectorStorage,
+        // Parallel inserts count.
         groups_count: usize,
-        points_remap: &[PointOffsetType],
         m: usize,
         m0: usize,
         ef: usize,
+        // If true, guarantee equality of result with CPU version for both single-threaded case.
+        // Required for tests.
         exact: bool,
+        // If points count is very big, we share visited flags buffer between multiple points.
+        // This parameter sets a factor how many points can share one visited flag.
         visited_flags_factor_range: std::ops::Range<usize>,
     ) -> OperationResult<Self> {
         let device = gpu_vector_storage.device();
-        let points_count = gpu_vector_storage.num_vectors();
+        let points_count = std::cmp::max(1, gpu_vector_storage.num_vectors());
```

Comment:
Why do we need at least 1 here?

Reply:
vulkan does not allow zero sized buffers allocation

Reply:
Would it be possible to somehow early return, and not create the GPU context all together?

Or is this just a safe guard that is in place if it ever accidentally happens so that we don't panic?

Reply:
Fixed. I don't create GPU context at all if the number of points is less than the minimum required points count to start GPU insertions (first points like in CPU are constructed on CPU)